runtime/src/main/java/org/corfudb/runtime/view/CorfuGuidGenerator.java

```diff
@@ -233,7 +234,17 @@ public long nextLong() {
      */
     private synchronized CorfuGuid getCorrectedTimestamp() {
         long currentTimestamp = System.currentTimeMillis();
-        if (currentTimestamp == previousTimestamp) {
+        if (currentTimestamp > previousTimestamp) {
+            final long oneYearInMillis = TimeUnit.DAYS.toMillis(365);
+            int maxCrazyNtpValueRetries = 1000;
+            while (currentTimestamp - previousTimestamp > oneYearInMillis) {
```

Comment:
should this sleep in between retries? how do we know this is "enough" cycles/time to typically correct itself?

Reply:
just experimental observations at this point - when ntp throws time around, it ends up correcting itself in subsequent calls. I just thought adding a sleep might slow down a problem that corrects itself quickly?

Reply:
ok. I was just wondering based on what we knew this was guaranteed to be enough time to recover... and one more question, what are the implications of this happening? is a warning enough?

Reply:
This loop will most likely execute way faster than any clock correction. I don't see how it addresses the time jump issue.

Reply:
logs had showed that in one call to currentTimeMillis() the value was something like INT_MAX and in the next call it had returned back to normal. I don't understand NTP to know why that happens, I can add a sleep if you think that helps.

Reply:
I don't feel like spinning on currentTimeMillis is robust. Why isn't it sufficient to just retry the whole transaction from a higher layer? All transactions are susceptible to non-conflict related aborts and need to retry anyways.

Reply:
we don't have control over that aspect right - I agree in general with the fact that they are supposed to retry the transaction, but according to them this part of the code should never throw any exception